File: regions/C++/regions/regions.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <opencv2/tracking.hpp>
#include <string>

cv::Mat loadImage(std::string pathImage);

void showImage(std::string title, cv::Mat& image);

int main(int argc, char** argv) {

    auto image = loadImage(argv[1]);
    
    auto rect2d = cv::selectROI(image);
    
    auto region = image(rect2d);
    
    region = 255 -region;
    
    showImage("image",image);



    return 0;

}

void showImage(std::string title, cv::Mat& image) {
    cv::namedWindow(title, cv::WINDOW_NORMAL);

    cv::imshow(title, image);
    cv::waitKey();
}

cv::Mat loadImage(std::string pathImage) {
    cv::Mat image = cv::imread(pathImage, cv::IMREAD_GRAYSCALE);

    if (not image.data) {
        std::cout << "imagem não carregada" << std::endl;
    }

    return image;
}

File: regions/Python/regions.py
import cv2
import sys


def main():
    image = cv2.imread(sys.argv[1], cv2.IMREAD_GRAYSCALE)

    r = cv2.selectROI("ROI", image)

    image[int(r[1]):int(r[1] + r[3]), int(r[0]):int(r[0] + r[2])] = 255 - image[int(r[1]):int(r[1] + r[3]),
                                                                          int(r[0]):int(r[0] + r[2])]

    cv2.imshow("region", image)
    cv2.waitKey(0)


if __name__ == '__main__':
    main()
